### PR TITLE
checker: skip generating operator if fit is cached for rule checker

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

If the fit is fetched from cache, the rule checker will skip checking instead of generating operator.

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: server/schedule/checker/rule_checker.go [hunk 1]
```diff
@@ -65,6 +66,16 @@ func (c *RuleChecker) Check(region *core.RegionInfo) *operator.Operator {
...
 // CheckWithFit checkWithFit is similar with Checker with placement.RegionFit
 func (c *RuleChecker) CheckWithFit(region *core.RegionInfo, fit *placement.RegionFit) *operator.Operator {
+	// If the fit is fetched from cache, it seems that the region doesn't need cache
+	if fit.IsCached() {
+		checkerCounter.WithLabelValues("rule_checker", "cache").Inc()
+		return nil
+	}
+
+	failpoint.Inject("assertCache", func() {
+		panic("shouldn't be here")
```
Comment (disksing) [inline server/schedule/checker/rule_checker.go:76]: maybe `RegionFit is not cached`

Comment (Yisaer) [reply server/schedule/checker/rule_checker.go]: updated.

**Review by disksing — APPROVED** [review]

**Pull request merged**